src/core/lib/surface/call.cc

```diff
@@ -2880,6 +2885,7 @@ void ClientPromiseBasedCall::PublishStatus(
         gpr_strdup(MakeErrorString(trailing_metadata.get()).c_str());
   }
   PublishMetadataArray(op_args.trailing_metadata, trailing_metadata.get());
+  recv_status_on_client_ = absl::monostate();
```

Comment:
Just for my understanding, what's the reason for this change?  It's not obvious to me how it relates to the rest of the PR.

Reply:
Oh bug fix that it showed up... we call into this function if we're in a state where there's an outstanding `RECV_STATUS_ON_CLIENT` op, but we should only complete it once... so if a secondary finish condition shows up later we want to ensure we're back in a different state before that happens.

Reply:
Okay, makes sense.  Can you please add a comment explaining that?

Reply:
done